src/main/java/com/example/nlearn/services/CustomOAuth2UserService.java

```diff
@@ -0,0 +1,34 @@
+package com.example.nlearn.services;
+
+
+import com.example.nlearn.models.User;
+import com.example.nlearn.models.CustomOAuth2User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.OAuth2Error;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+
```

Comment:
We don't leave two newlines after import statements (also above)

Reply:
done